silx/gui/plot/backends/BackendMatplotlib.py

```diff
@@ -402,9 +402,13 @@ def addCurve(self, x, y, legend,
                                    s=symbolsize**2)
             artists.append(scatter)
 
-            if fill:
+            if fill or baseline is not None:
```

Comment:
I would keep `fill` and `baseline` independent, i.e., if `fill` is `False`, then it is not filled even if there is a `baseline`.